ttnmqtt/ttnmqtt.py

```
@@ -0,0 +1,184 @@
+import grpc
+import os
+
+os.environ['GRPC_SSL_CIPHER_SUITES'] = ("ECDHE-ECDSA-AES256-"
+                                        "GCM-SHA384:ECDHE-RSA-AES256-"
+                                        "GCM-SHA384:ECDHE-ECDSA-CHACHA20-"
+                                        "POLY1305:"
+                                        "ECDHE-RSA-CHACHA20-POLY1305:"
+                                        "ECDHE-ECDSA-AES128-GCM-SHA256:"
+                                        "ECDHE-RSA-AES128-GCM-SHA256:"
+                                        "ECDHE-ECDSA-AES256-SHA384:"
+                                        "ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-"
+                                        "AES128-SHA256:"
+                                        "ECDHE-RSA-AES128-SHA256")
+
+
+def _json_object_hook(d):
+    return namedtuple('MSG', d.keys())(*d.values())
+
+
+def json2obj(data):
+    return json.loads(data, object_hook=_json_object_hook)
+
+
+class DownlinkMessage:
+    def __init__(self, port, confirmed=False, schedule=None):
+        self.port = port
+        if confirmed:
+            self.confirmed = confirmed
+        if schedule:
+            self.schedule = schedule
+
+    def obj2json(self):
+        json_msg = json.dumps(self.__dict__)
+        return str(json_msg)
+
+
+class MyEvents(Events):
+    __events__ = (
+        "uplink_msg",
+        "downlink_msg",
+        "connect",
+        "close")
+
+
+class MQTTClient:
+
+    def __init__(self, appID, appAccessKey, mqttAddress=""):
+        self.__client = mqtt.Client()
+        self.__appID = appID
+        self.__accessKey = appAccessKey
+        self.__events = MyEvents()
+        self.__mqttAddress = mqttAddress
+        self.__ErrorMsg = ""
+        self.connectFlag = 2
+        self.disconnectFlag = 2
+        self.midCounter = 0
+        try:
+            self.connect()
+            self.start()
+        except:
+            self.__ErrorMsg = "Connection failed"
```

+ The exception should be used to provide more information in the error message
+ Is `__ErrorMsg` documented?

IMHO it could be good to just let the exception be dropped, and document where exceptions can happen.